homeassistant/components/govee_light_api/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Coordinator for Govee Local API."""
+
+from collections.abc import Callable
+from datetime import timedelta
+import logging
+
+from govee_local_api import GoveeController, GoveeDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_DISCOVERY_INTERVAL_DEFAULT,
+    CONF_LISENING_PORT_DEFAULT,
+    CONF_MULTICAST_ADDRESS_DEFAULT,
+    CONF_TARGET_PORT_DEFAULT,
+)
+
+
+class GoveeLocalApiCoordinator(DataUpdateCoordinator):
+    """Govee Local API coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        scan_interval: timedelta,
+        logger: logging.Logger,
+    ) -> None:
+        """Initialize my coordinator."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name="GoveeLightLocalApi",
+            update_interval=scan_interval,
+        )
+
+        self._controller = GoveeController(
+            loop=hass.loop,
+            logger=logger,
+            broadcast_address=CONF_MULTICAST_ADDRESS_DEFAULT,
+            broadcast_port=CONF_TARGET_PORT_DEFAULT,
+            listening_port=CONF_LISENING_PORT_DEFAULT,
+            discovery_enabled=True,
+            discovery_interval=CONF_DISCOVERY_INTERVAL_DEFAULT,
+            discovered_callback=None,
+            update_enabled=False,
+        )
+
+    async def start(self) -> None:
+        """Start the Govee coordinator."""
+        await self._controller.start()
+        self._controller.send_update_message()
+
+    async def set_discovery_callback(
+        self, callback: Callable[[GoveeDevice, bool], bool]
+    ) -> None:
+        """Set discovery callback for automatic Govee light discovery."""
+        self._controller.set_device_discovered_callback(callback)
+
+    def clenaup(self) -> None:
+        """Stop and clenaup the cooridinator."""
+        self._controller.clenaup()
```

Comment:
It's cleanup. Please fix this in the code or it'll probably stay here forever

Reply:
I was able to get it wrong also in the library 😓 I'll also fix there